### PR TITLE
[Sundials] update for sundials 5.2

### PR DESCRIPTION
**Pull request by jd-lara**
cc. @ViralBShah

### CODE REVIEW

File: S/Sundials@5/build_tarballs.jl [hunk 1]
```diff
@@ -16,7 +16,7 @@ cd $WORKSPACE/srcdir/sundials*
 
 # Set up CFLAGS
 if [[ "${target}" == *-mingw* ]]; then
-    atomic_patch -p1 $WORKSPACE/srcdir/patches/Sundials_windows.patch
+    # atomic_patch -p1 $WORKSPACE/srcdir/patches/Sundials_windows.patch
     # Work around https://github.com/LLNL/sundials/issues/29
     export CFLAGS="${CFLAGS} -DBUILD_SUNDIALS_LIBRARY"
```
Comment (giordano) [inline S/Sundials@5/build_tarballs.jl]: Also this `export` shouldn't be necessary, according to https://github.com/LLNL/sundials/issues/29#issuecomment-606679985.  It's probably useful to see whether that's the case

Comment (jd-lara) [reply S/Sundials@5/build_tarballs.jl]: Agreed, I just wanted to go one change at a time in case it breaks.

Comment (ViralBShah) [reply S/Sundials@5/build_tarballs.jl]: The windows builds are failing so the patch probably is needed. I thought the upstream issue referred to not needing the -DBUILD_SUNDIALS_LIBRARY. I don't think we filed an issue for the patch we are carrying.

Comment (jd-lara) [reply S/Sundials@5/build_tarballs.jl]: Trying again

Comment (ViralBShah) [reply S/Sundials@5/build_tarballs.jl]: I guess it did not work.

Comment (jd-lara) [reply S/Sundials@5/build_tarballs.jl]: Now it does. We can merge so I can use this in Sundials.jl

**Pull request merged**